src/NUnitFramework/framework/Internal/Tests/TestAssembly.cs

```diff
@@ -39,25 +39,25 @@ public class TestAssembly : TestSuite
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="TestAssembly"/> class
-        /// specifying the Assembly and the path from which it was loaded.
+        /// specifying the Assembly and the name of the assembly.
         /// </summary>
         /// <param name="assembly">The assembly this test represents.</param>
-        /// <param name="path">The path used to load the assembly.</param>
-        public TestAssembly(Assembly assembly, string path)
-            : base(path)
+        /// <param name="name">The name of the assembly.</param>
+        public TestAssembly(Assembly assembly, string name)
+            : base(name)
         {
             this.Assembly = assembly;
-            this.Name = Path.GetFileName(path);
+            this.Name = name;
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TestAssembly"/> class
-        /// for a path which could not be loaded.
+        /// specifying the name of the assembly.
         /// </summary>
-        /// <param name="path">The path used to load the assembly.</param>
-        public TestAssembly(string path) : base(path)
+        /// <param name="name">The name of the assembly.</param>
+        public TestAssembly(string name) : base(name)
```

Comment:
I'd almost ask for this to be `suiteName` too for full clarity, but it seems more consistent to align with the base class's parameter name and property.